simplemonitor/simplemonitor.py

```diff
@@ -211,10 +215,16 @@ def _load_monitors(self, filenames: Sequence[Union[Path, str]]) -> None:
             new_monitor = cls(this_monitor, config_options)
             # new_monitor.set_mon_refs(m)
 
-            module_logger.info(
-                "Adding %s monitor %s: %s", monitor_type, this_monitor, new_monitor
-            )
-            self.add_monitor(this_monitor, new_monitor)
+            if new_monitor.enabled:
+                module_logger.info(
+                    "Adding %s monitor %s: %s", monitor_type, this_monitor, new_monitor
+                )
+                self.add_monitor(this_monitor, new_monitor)
+            else:
+                module_logger.info(
+                    "Skipping disabled %s monitor: %s", monitor_type, this_monitor
```

Comment:
Have you tested this with dependencies between monitors? I forget how that code goes exactly but there's a chance that not loading a disabled monitor which is referred to in a dep will result in a validation error or exception later.

I also like disabled monitors showing up in e.g. the HTML output to remind me that there's a check which is not being run currently.

Reply:
I have checked now and the program breaks with "Configuration error: dependency %s of monitor %s is not defined!".

Is it not actually a good thing? You should not rely on disabled monitors - ie. something with unknown status.

I have tried both Alerters and Loggers with enabled=false chained with depend=... - and the program does not break.

Though it would be nice to have some logging info about depending on disabled items...

Reply:
Maybe a more precise text would do:

      "Configuration error: dependency %s of monitor %s is not defined or is not enabled!",

Reply:
I'll take this change and see how it goes with my own usage - I have a disabled monitor right now for a thing I want to remember isn't being monitored properly (due to an external issue) so it shows up in my HTML status page.